pkg/plugins/resources/k8s/native/pkg/test/within.go

```diff
@@ -0,0 +1,25 @@
+package test
+
+import (
+	"time"
+
+	"github.com/onsi/gomega"
+)
+
+// Within returns a function that executes the given test task in
+// a dedicated goroutine, asserting that it must complete within
+// the given timeout.
+//
+// See https://github.com/onsi/ginkgo/blob/v2/docs/MIGRATING_TO_V2.md#removed-async-testing
+func Within(timeout time.Duration, task func()) func() {
+	return func() {
+		done := make(chan interface{})
+
+		go func() {
```

Comment:
You are missing here
```
defer ginkgo.GinkgoRecover()
```

Try to write a simple test with failing assertion
```
var _ = Describe("", func() {
	It("", test.Within(1 * time.Second, func() {
		Expect(true).To(BeFalse())
	}))
})
```
to see a difference

Reply:
Great catch!

I added this and tried some simple tests to ensure that it works, and to verify the `Within` timeout. I couldn't figure out how to get ginkgo to let me do and `expected-fail` test, but if there's a way we can add that.